presto-hive/src/test/java/com/facebook/presto/hive/TestHiveSplitSource.java

```diff
@@ -254,6 +260,40 @@ public void testEmptyBucket()
         assertEquals(getSplits(hiveSplitSource, OptionalInt.of(3), 10).size(), 0);
     }
 
+    @Test
+    public void testLoadAllSplitsBeforeScheduling()
+            throws Exception
+    {
+        ConnectorSession session = new TestingConnectorSession(
+                new HiveSessionProperties(
+                        new HiveClientConfig().setLoadAllSplitsBeforeSchedulingForGroupedExecution(true),
+                        new OrcFileWriterConfig(),
+                        new ParquetFileWriterConfig()).getSessionProperties());
+        HiveSplitSource hiveSplitSource = HiveSplitSource.bucketed(
+                session,
+                "database",
+                "table",
+                TupleDomain.all(),
+                10,
+                10,
+                new DataSize(1, MEGABYTE),
+                new TestingHiveSplitLoader(),
+                Executors.newFixedThreadPool(5),
+                new CounterStat());
+        for (int i = 0; i < 10; i++) {
+            hiveSplitSource.addToQueue(new TestSplit(i, OptionalInt.of(0)));
+            assertEquals(hiveSplitSource.getBufferedInternalSplitCount(), i + 1);
+        }
+        Future<ConnectorSplitBatch> batch = hiveSplitSource.getNextBatch(new HivePartitionHandle(0), 10);
+        MILLISECONDS.sleep(200);
+        assertFalse(batch.isDone());
```

Comment:
Add a comment about the schedule cannot start since `noMoreSplit` signal is not yet sent.